catroid/src/org/catrobat/catroid/physics/PhysicsLook.java

```diff
@@ -128,10 +133,26 @@ public void setRotation(float degrees) {
 
 	@Override
 	public void setScale(float scaleX, float scaleY) {
+		Vector2 oldScales = new Vector2(getScaleX(), getScaleY());
+		if (scaleX < 0.0f || scaleY < 0.0f) {
+			scaleX = 0.0f;
+			scaleY = 0.0f;
+		}
+
+		int scaleXComp = Math.round(scaleX * SCALEFACTOR_ACCURACY);
+		int scaleYComp = Math.round(scaleY * SCALEFACTOR_ACCURACY);
+		if (scaleXComp == Math.round(oldScales.x * SCALEFACTOR_ACCURACY) && scaleYComp == Math.round(oldScales.y * SCALEFACTOR_ACCURACY)) {
+			return;
+		}
+
+		scaleX = scaleXComp / SCALEFACTOR_ACCURACY;
+		scaleY = scaleYComp / SCALEFACTOR_ACCURACY;
 		super.setScale(scaleX, scaleY);
```

Comment:
I get what you're saying but the problem is that re-scaling physicsObjects uses the scaleX variable of the look object (defined in Actor -> Widget -> Image -> Look) so there are no separate scale values for physicsLooks and the values passed to super.setScale are the exact same which are used by the physics engine.

Should I just call super.setScale with the original values and remove the lines from 142 to 149 ? I don't think allowing scales of a negative value would be a good idea and would therefore keep lines 136-140.